### ISSUE TITLE
[gatsby-theme-minimal-blog]: Failing to install the theme

### ISSUE THREAD

**Issue by cesards** [LekoArts/gatsby-themes#595]
## Description

I'm getting an error when running:
```
npm install @lekoarts/gatsby-theme-minimal-blog
```
It seems it could be related to needing to update a few dependencies (react):
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR!   peer react@"^16.4.2 || ^17.0.0" from gatsby@2.32.13
npm ERR!   node_modules/gatsby
npm ERR!     peer gatsby@"^2.13.3" from @lekoarts/gatsby-theme-minimal-blog@2.7.6
npm ERR!     node_modules/@lekoarts/gatsby-theme-minimal-blog
npm ERR!       @lekoarts/gatsby-theme-minimal-blog@"*" from the root project
npm ERR!   1 more (react-dom)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.9.0" from @lekoarts/gatsby-theme-minimal-blog@2.7.6
npm ERR! node_modules/@lekoarts/gatsby-theme-minimal-blog
npm ERR!   @lekoarts/gatsby-theme-minimal-blog@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See ~/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2021-05-31T21_46_46_081Z-debug.log
```

### Steps to reproduce

Start a fresh theme install.

### Expected result

NPM should install all the dependencies from the theme.

### Actual result

It fails.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here:
```
  System:
    OS: macOS 11.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 3.2.2 - /usr/local/bin/fish
  Binaries:
    Node: 16.2.0 - /usr/local/bin/node
    npm: 7.13.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 90.0.4430.212
    Firefox: 85.0.1
    Safari: 14.1
  npmGlobalPackages:
    gatsby-cli: 3.6.0
```

**Comment by emptypockets-dev** [comment]
I'm having the same issue. Looks like it is dependency related. I'm happy to try things if you have any suggestion. I'll start with checking versions for Babel related stuff.

**Comment by cesards** [comment]
Solved by running:
```
npm install react@16.9.0
npm install react-dom@16.4.2
```
Also, it's worth mentioning all these deprecated dependencies:
```
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
```

I have a few questions:

1. Would these warnings go away if we updated [this file](https://github.com/LekoArts/gatsby-themes/blob/master/package.json)? (having in consideration there aren't breaking changes).

2. What's the procedure we should follow to keep our site up to the base theme (`@lekoarts/gatsby-theme-minimal-blog`)? Should we just run `npm update @lekoarts/gatsby-theme-minimal-blog` or `npm install x@latest` would be preferred?

**Comment by emptypockets-dev** [comment]
@cesards That unfortunately did not work for me.

**Comment by cesards** [comment]
I think the theme is broken anyways. It doesn't build the site when running `gatsby develop` ([this issue](https://github.com/LekoArts/gatsby-themes/issues/593)):
```
 ~/W/W/cesards (master)> gatsby develop

 ERROR

(node:10101) [DEP_WEBPACK_LIBRARY_TEMPLATE_PLUGIN] DeprecationWarning: webpack.LibraryTemplatePlugin is deprecated and has been replaced by compilation.outputOptions.library or compilation.addEntry + passing a library option
(Use `node --trace-deprecation ...` to show where the warning was created)


 ERROR

(node:10101) [DEP_WEBPACK_AGGRESSIVE_SPLITTING_PLUGIN] DeprecationWarning: AggressiveSplittingPlugin is deprecated in favor of SplitChunksPlugin

success open and validate gatsby-configs - 0.105s
success load plugins - 3.599s
success onPreInit - 0.042s
success initialize cache - 0.010s
success copy gatsby files - 0.084s
success onPreBootstrap - 0.024s
success createSchemaCustomization - 0.023s
success Checking for changed pages - 0.001s
success source and transform nodes - 0.799s

 ERROR

Missing onError handler for invocation 'building-schema', error was 'Error: Cannot create as TypeComposer the following value: ImageFormat.'. Stacktrace was 'Error: Cannot create as TypeComposer the following value: ImageFormat.
```

**Comment by LekoArts** [comment]
Hi!

TL;DR see this SO for a longer explanation: https://stackoverflow.com/a/66620869/10164092

The theme is on the latest Gatsby v2 version (and the v3 release is under its way: https://github.com/LekoArts/gatsby-themes/issues/570) and thus has some older dependencies in them that declared React 16 as a peerDependency. npm 7 by default will try to be "clever" and install those. You'll want to keep the old behavior and use the `--legacy-peer-deps` flag. Unfortunately we also saw this in the v2 => v3 migration (https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#update-gatsby-version) so maybe going forward I'll change up the install instructions and add the flag by default.

The deprecated packages you mentioned are out of my control, those are all Gatsby dependencies. Updating to v3 should help there, too.

**Comment by cesards** [comment]
Hey @LekoArts, thanks for taking the time to provide more context on the issue. It makes sense.

I tried what you said on the other issue and I couldn't make it work: https://github.com/LekoArts/gatsby-themes/issues/596#issuecomment-852123250

I'll try using `gatsby-starter` to create a new site from the scratch, moving stuff I previously had. As a non-web developer, this might be a good exercise for me so I figure some stuff out. I'll be paying attention to your updates. Please let me know if there's anything else I could do to help,

Cesar

**Comment by Ryunaq** [comment]
I had this problem too, the v3 update to gatsby didn't fixed it. Since I wanted to use the starter by using `gatsby new` then the npm `--legacy-peer-deps` flag didn't help. For reference to anyone else, what did help was install globally gatsby using yarn, and setting it as its default package manager, see https://www.gatsbyjs.com/docs/glossary/yarn/ .

If you want I can do a commit to the minimal-blog theme @LekoArts , also since I am already here, thanks for the great themes!
